src/raft.c

```diff
@@ -577,10 +579,10 @@ static void handleEndClientSession(RedisRaftCtx *rr, raft_entry_t *entry)
 
 static void clearClientSessions(RedisRaftCtx *rr)
 {
-    void *client_session;
+    ClientSession *client_session;
     RedisModuleDictIter *iter = RedisModule_DictIteratorStartC(rr->client_session_dict, "^", NULL, 0);
-
     while (RedisModule_DictNextC(iter, NULL, (void **) &client_session) != NULL) {
+        RedisModule_DeauthenticateAndCloseClient(rr->ctx, client_session->client_id);
```

Comment:
I think we should consider log replay and free clients only if we are in the current term? Otherwise, we may free unrelated clients, no?

Reply:
Also, we may disconnect unrelated clients on followers this way. 
Maybe similar to you did, we need to store term in the session object and disconnect clients if the `NOOP term = session_term + 1`.

Reply:
so I think its impossible to have a "client_session" (i.e. client_id -> client_session" mapping) that will be in the current term.

why?

the way we create a client_session mapping is a WATCH command on the log being applied.  

Any WATCH commands appended before the NO_OP will be on the old term.
Any WATCH commands appended after the NO_OP will be on the new term.

Any sessions created by old WATCH commands before the NO_OP will be freed/closed at NO_OP time.  This won't impact clients that issued them after the NO_OP was appended (but perhaps before it was applied) because they will come after in the log. 

Basically, I think its impossible for their to be client_sessions (at NO_OP) time that exist for the new term.  

The only way this would be wrong is if its possible for their to be log entries for a the NO_OP's term before the NO_OP, which as I said, I think is impossible (i.e. would break raft guarantees).